### PR TITLE
[SPARK-34111][BUILD] Deconflict the jars jakarta.servlet-api-4.0.3.jar and javax.servlet-api-3.1.0.jar

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

Use the right name for the artifact `javax.servlet-api` in hadoop 3.2 profile

### Why are the changes needed?

fix deps issue

### Does this PR introduce _any_ user-facing change?

no
### How was this patch tested?

1. should pass existing jenkins
2. I have manually tested a spark application with the `javax.servlet-api-3.1.0.jar` deleted and it worked fine

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @HyukjinKwon @dongjoon-hyun thanks for reviewing

File: dev/deps/spark-deps-hadoop-3.2-hive-2.3 [hunk 1]
```diff
@@ -126,7 +126,6 @@ janino/3.0.16//janino-3.0.16.jar
 javassist/3.25.0-GA//javassist-3.25.0-GA.jar
 javax.inject/1//javax.inject-1.jar
 javax.jdo/3.2.0-m3//javax.jdo-3.2.0-m3.jar
-javax.servlet-api/3.1.0//javax.servlet-api-3.1.0.jar
```
Comment (dongjoon-hyun) [inline dev/deps/spark-deps-hadoop-3.2-hive-2.3:129]: Oh, nice.

**Comment by dongjoon-hyun** [comment]
Thank you, @yaooqinn . Yes, let's see the test result.

**Comment by HyukjinKwon** [comment]
Nice, thank you @yaooqinn!

**Review by HyukjinKwon — COMMENTED** [review]
The fix itself looks good.  should be good to go if tests pass

**Comment by yaooqinn** [comment]
retest this please

**Comment by yaooqinn** [comment]
```
- decommission workers ensure that shuffle output is regenerated even with shuffle service *** FAILED *** (20 seconds, 783 milliseconds)
[info]   5 did not equal 4 Expected 4 tasks but got List(0:0:0:0-SUCCESS, 0:0:1:0-FAILED, 0:0:1:1-SUCCESS, 0:1:0:0-SUCCESS, 1:0:0:0-SUCCESS) (DecommissionWorkerSuite.scala:190)
[info]   org.scalatest.exceptions.TestFailedException:
[info]   at org.scalatest.Assertions.newAssertionFailedException(Assertions.scala:472)
[info]   at org.scalatest.Assertions.newAssertionFailedException$(Assertions.scala:471)
[info]   at org.scalatest.Assertions$.newAssertionFailedException(Assertions.scala:1231)
[info]   at org.scalatest.Assertions$AssertionsHelper.macroAssert(Assertions.scala:1295)
[info]   at org.apache.spark.deploy.DecommissionWorkerSuite.$anonfun$new$6(DecommissionWorkerSuite.scala:190)
```

seems just a flaky test itself and not related to this change

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
retest this please

**Review by HyukjinKwon — APPROVED** [review]

**Comment by yaooqinn** [comment]
look ok in both mvn and sbt tests

**Comment by HyukjinKwon** [comment]
Hm, wait. Let's wait for GA results. Seems like this particular job (which is related):

```
Build modules: streaming, sql-kafka-0-10, streaming-kafka-0-10, mllib-local, mllib, yarn, mesos, kubernetes, hadoop-cloud, spark-ganglia-lgpl (JDK 8, hadoop3.2, hive2.3)
```

keeps pending.

**Comment by yaooqinn** [comment]
retest this please

**Review by srowen — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Sorry, but could you rebase once more, @yaooqinn ?

**Comment by yaooqinn** [comment]
yes, I checked and saw the `servlet-api` gone away, we don't need this PR anymore @dongjoon-hyun.

**Comment by dongjoon-hyun** [comment]
Thank you for the confirmation, @yaooqinn .

**Comment by dongjoon-hyun** [comment]
Instead, could you open a PR to branch-3.1, @yaooqinn ?

**Comment by dongjoon-hyun** [comment]
SPARK-33212 is only applied at `master` branch for Apache Spark 3.2.0. So, SPARK-34111 is still a blocker for Apache Spark 3.1.0 release.

**Comment by yaooqinn** [comment]
OK

**Pull request closed without merging**